### DISCUSSION TITLE
Is there way to Listen the the event of workflow instance completion in elsa ?

### DISCUSSION THREAD

**Discussion by dwevedivaibhav** [elsa-workflows/elsa-core#5575] [Design & tech]
Hi @sfmskywalker ,

I have use case like have parent and child workflow  i wanted to update some meta info in my table before completion of workflow , Is there is way to do that?

Can we have any event which can launch during starting and before or completion of workflow?

**Comment by glime-ai** [comment]
To update meta information in your table before the completion of a workflow in Elsa, you can use custom activities or events that are triggered at specific points in your workflow. Elsa does not have built-in events that are specifically designed to run before the completion of a workflow, but you can design your workflow in such a way that it includes activities that perform the necessary updates right before the workflow is set to complete.

Here are some steps you can follow:

1. **Custom Activities**: Create a custom activity that updates your meta information in the database. Place this activity in your workflow at the point where you want the update to occur, which would be right before the workflow's completion.

2. **Events**: Use the `Finish` activity to mark the end of your workflow. Before this activity, you can add your custom activity that performs the update.

3. **Workflow Execution Events**: Elsa provides a way to handle workflow execution events such as `WorkflowStarting`, `WorkflowStarted`, `WorkflowCompleted`, etc. You can subscribe to these events and execute your custom logic to update the meta information accordingly.

4. **Middleware**: Although not directly related to workflow events, you can also use ASP.NET Core middleware to intercept HTTP requests that trigger workflows and perform actions before and after the workflow execution.

5. **Persistence Behavior**: Ensure that your workflow's persistence behavior is set to persist workflow instances before completion if you need the meta information to be updated in the database before the workflow instance is marked as complete.

6. **Child Workflow**: If you are using a child workflow, you can use the `DispatchWorkflow` activity in the parent workflow to start the child workflow and wait for its completion. Before the `DispatchWorkflow` activity, you can add an activity to update your meta information.

Remember to publish your workflow after making changes to ensure that the latest version is executed.

If you need further clarification or assistance with implementing any of these suggestions, please provide more details about your specific use case and the meta information you are trying to update.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
